share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -353,6 +363,93 @@ function(ExternalZephyrProject_Cmake)
   import_kconfig(CONFIG_ ${BINARY_DIR}/zephyr/.config TARGET ${ZCMAKE_APPLICATION})
 endfunction()
 
+# Usage:
+#   ExternalZephyrProject_Group(<name>
```

Comment:
Not sure about the `_Group` name here.

The `ExternalZephyrProject_Group()` seems to more handle properties and not be doing an explicit step as described:
https://cmake.org/cmake/help/latest/module/ExternalProject.html#explicit-step-management

That's a difference to the `ExternalZephyrProject_CMake()` function which actually carries out a dedicate step, which is the CMake configure step, but with the slight difference that the step is executed at sysbuild CMake configure time and not sysbuild build time (which would be the normal behavior of an external step).

Reply:
also see this comment: https://github.com/zephyrproject-rtos/zephyr/pull/57884#discussion_r1201620427

I believe groups should be independent of `ExternalZephyrProject` cause we want to allow handling of non-Zephyr projects.
As long as such project can provide / generate a runners we can still use west to flash them.

This indicates that the group feature should be independent of the `ExternalZephyrProject` feature.

Reply:
Continued: https://github.com/zephyrproject-rtos/zephyr/pull/58174#discussion_r1201660043